### ISSUE TITLE
Contributing?

### ISSUE THREAD

**Issue by brianespinosa** [whoisryosuke/next-mdx-deck#5]
Hi @whoisryosuke!

I really love what you have put together here so far. 

Yesterday I was working on packaging up a theme for my organization to use with [mdx-deck](https://github.com/jxnblk/mdx-deck) and I was surprisingly frustrated trying to get some of the theming to work. Many changes required killing build processes and starting them over again. I think this has to do with the caching in Gatsby... which comes with its own issues (all of the install warnings with dependencies, etc). There are a lot of things that feel like they should get updated with mdx-deck to make theming and configuration easier. I got to thinking that it would potentially be easier to build a better solution with Next instead. Which is how I found your project this morning. 👍🏽 

Are you interested in having contributions to this project? If so, we should add a CONTRIBUTING.md to the project and define some vision and direction. If you prefer this to be a personal project only, that is okay too.

I did make a fork and tested out a few things in terms of starting to centralize configuration. I think it would be great if the project was self contained and you could just point it at an MDX file and have a slide spin up. That way this could eventually be published as a package. If all it needed was a deck.config.js, deck.mdx, and possibly a deckTheme.js, that would be awesome.

Let me know what you think. I'd be happy to share some ideas in terms of abstracting out some of the implementation details into config the way you already started.

**Comment by whoisryosuke** [comment]
Hey @brianespinosa 👋🏼

Thank you for the kind words and interest! 🔥🤘🏼

Always down for contributions 👍🏼 I can add a CONTRIBUTING.MD to help start this process.

Not sure if there's a particular vision for the project, I have a roadmap I initially outlined for features, but it's definitely not concrete. [You can see it here under v2.](https://www.notion.so/whoisryosuke/Next-MDX-Deck-a75de9bf35df4a1fb4dc349a1dadb629). It's basically smaller features to reach parity with mdx-deck and a couple quality of life improvements.

I like the idea of CLI-izing the project to make it easier to consume as an end user, really helps encapsulate all the Next logic. And like you said, would allow for more simpler config based setup. This seems fairly achievable, especially looking at the [mdx-deck CLI](https://github.com/jxnblk/mdx-deck/blob/master/packages/mdx-deck/cli.js).

Feel free to throw in a PR if you have any suggestions, always open to feedback and refactors. As long as the app works and gets better for everyone, it's all good 😁 If you'd prefer to put in a more conceptual PR, that's also cool too!

I'm personally paused on this project for a little bit, my focus is on a couple other things right now, but I'll be circling back to this soon when I start recording some talks again 🎙✅

If you have any questions about anything feel free to hit me up or create a new issue 👍🏼

**Comment by whoisryosuke** [comment]
@brianespinosa [Created a branch for the CLI.](https://github.com/whoisryosuke/next-mdx-deck/tree/cli) Feel free to test it out 👍

Should be able to clone the branch, `npm link` it to a project, and run the CLI (using the new README). 

I'm going to stress test it locally then consider a v1 release. Gotta pick up a user's `deck.config.js` to replace the slide layout data and whatnot, but the core functionality should be good to go.

**Comment by DandyDev** [comment]
Hey there! I'm evaluating code-based presentation tools to build my future presentations in. I checked out Marp and RevealJS, but was ultimately drawn towards using MDX to build slides.
First I found mdx-deck, but then I also found this cool project. I'm actually tempted to start using this one, but the one thing that is really missing, is a good CLI to make it super easy to get started. Are you planning to merge and publish your CLI branch @whoisryosuke ?
